lib/dialects/mysql/index.js

```diff
@@ -91,7 +91,7 @@ function connectionErrorHandler(client, connection, err) {
   if (connection && err && err.fatal) {
     if (connection.__knex__disposed) return;
     connection.__knex__disposed = true;
-    client.pool.genericPool.destroy(connection);
+    client.pool.destroy(connection);
```

Comment:
.release releases the resource to the pool
.remove removes the resource from the pool nicely
.destroy removes the resource from the pool dirtily

Knex only passes a remove handler so far so you should always call .remove